Algorithm.Framework/Selection/UncorrelatedUniverseSelectionModel.py

```diff
@@ -76,12 +76,13 @@ def SelectCoarse(self, algorithm, coarse):
             if not data.IsReady:
                 newSymbols.append(symbol)
 
-        # Warm up the dictionary objects of selected symbols and benchmark that do not have enought data
+        # Warm up the dictionary objects of selected symbols and benchmark that do not have enough data
         if len(newSymbols) > 1:
             history = algorithm.History(newSymbols, self.historyLength, Resolution.Daily)
-            history = history.close.unstack(level=0)
-            for symbol in newSymbols:
-                self.cache[symbol].Warmup(history)
+            if not history.empty:
+                history = history.close.unstack(level=0).fillna(method='bfill')
```

Comment:
We cannot use backward fill, `bfill`, to warm up indicators, because it will introduce look-ahead bias. Instead, we should use forward fill, `ffill`, and drop all remaining `nan`:
```python
history = history.close.unstack(level=0).ffill().dropna()
```